types/jsx.d.ts

```diff
@@ -36,7 +36,7 @@ export interface CSSProperties
    * For examples and more information, visit:
    * https://github.com/frenic/csstype#what-should-i-do-when-i-get-type-errors
    */
-  [v: `--${string}`]: string | number | undefined
+  [v: '--${string}']: string | number | undefined
```

Comment:
This looks like intentional from devs since you're using a string '' and not template string `` you can specify whatever you want as a key the previous behaviour will enforce you to use a key like this '--key'.